integrations/jina/src/jina_haystack/document_embedder.py

```diff
@@ -0,0 +1,183 @@
+                             to keep the logs clean.
+        :param metadata_fields_to_embed: List of meta fields that should be embedded along with the Document text.
+        :param embedding_separator: Separator used to concatenate the meta fields to the Document text.
+        """
+        # if the user does not provide the API key, check if it is set in the module client
+        if api_key is None:
+            try:
+                api_key = os.environ["JINA_API_KEY"]
+            except KeyError as e:
+                msg = (
+                    "JinaDocumentEmbedder expects a Jina API key. "
+                    "Set the JINA_API_KEY environment variable (recommended) or pass it explicitly."
+                )
+                raise ValueError(msg) from e
+
+        self.model_name = model_name
+        self.prefix = prefix
+        self.suffix = suffix
+        self.prefix = prefix
+        self.suffix = suffix
+        self.batch_size = batch_size
+        self.progress_bar = progress_bar
+        self.metadata_fields_to_embed = metadata_fields_to_embed or []
+        self.embedding_separator = embedding_separator
+        self._session = requests.Session()
+        self._session.headers.update(
+            {
+                "Authorization": f"Bearer {api_key}",
+                "Accept-Encoding": "identity",
+                "Content-type": "application/json",
+            }
+        )
+
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {"model": self.model_name}
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        This method overrides the default serializer in order to avoid leaking the `api_key` value passed
+        to the constructor.
+        """
+        return default_to_dict(
+            self,
+            model_name=self.model_name,
+            prefix=self.prefix,
+            suffix=self.suffix,
+            batch_size=self.batch_size,
+            progress_bar=self.progress_bar,
+            metadata_fields_to_embed=self.metadata_fields_to_embed,
+            embedding_separator=self.embedding_separator,
+        )
+
+    def _prepare_texts_to_embed(self, documents: List[Document]) -> List[str]:
+        """
+        Prepare the texts to embed by concatenating the Document text with the metadata fields to embed.
+        """
+        texts_to_embed = []
+        for doc in documents:
+            meta_values_to_embed = [
+                str(doc.meta[key])
+                for key in self.metadata_fields_to_embed
+                if key in doc.meta and doc.meta[key] is not None
+            ]
+            text_to_embed = (
+                self.prefix + self.embedding_separator.join([*meta_values_to_embed, doc.content or ""]) + self.suffix
+            )
+
+            text_to_embed = text_to_embed.replace("\n", " ")
```

Comment:
This is recommended for OpenAI embedding models.
Jina models require the same?

Reply:
Removed, thanks